jakarta jaxws namespaces update

senivam opened the pull request:
Signed-off-by: Maxim Nesen <maxim.nesen@oracle.com>

senivam commented:
changing jaxws namespaces according to the [coresponing change](https://github.com/eclipse-ee4j/metro-jax-ws/pull/163) in jaxws-ri

lukasj commented on src/com/sun/ts/tests/jaxws/wsa/w2j/document/literal/typesubstitution/custom-client.xml:
```diff
@@ -18,7 +18,7 @@
+         xmlns="https://jakarta.ee/xml/ns/jaxws"
          xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" 
          xmlns:xsd="http://www.w3.org/2001/XMLSchema" 
          xmlns:jaxb="http://java.sun.com/xml/ns/jaxb">
```
`xmlns:jaxb="http://java.sun.com/xml/ns/jaxb` needs to be changed to `xmlns:jaxb="https://jakarta.ee/xml/ns/jaxb` too

RohitKumarJain commented:
I observed for the recently updated files in this PR, Copyright (c) year is still not updated, ex: src/com/sun/ts/tests/webservices13/ejb/annotations/WSEjbWSRefRespBindAndAddressingCombinedTest/custom-server.xml , src/com/sun/ts/tests/webservices13/ejb/annotations/WSEjbPkgInWarUnderWebInfLibTest/customfile.xml 

Can you please check and update the Copyright (c) year, in the modified files? Thanks

lukasj reviewed (approved):
LGTM. Note that JAX-WS acronym should be changed to 'Jakarta XML Web Services' but that can be done in other PR. Thanks!

alwin-joseph commented on src/com/sun/ts/tests/jaxws/ee/w2j/rpc/literal/httptest/jakartaEE.webservices.xml:
```diff
@@ -17,11 +17,11 @@
              version="1.2"
 	     xmlns:wsdl="http://httptestservice.org/wsdl"
-             xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/javaee_web_services_1_2.xsd">
+             xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/javaee_web_services_1_2.xsd">
```
should be https://jakarta.ee/xml/ns/jakartaee/jakartaee_web_services_2_0.xsd

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/jaxws/ee/w2j/rpc/literal/sec/secbasic/jakartaEE.webservices.xml:
```diff
@@ -17,11 +17,11 @@
              version="1.2"
 	     xmlns:wsdl="http://BasicAuthServiceTestService.org/wsdl"
-             xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/javaee_web_services_1_2.xsd">
+             xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/javaee_web_services_1_2.xsd">
```
should be https://jakarta.ee/xml/ns/jakartaee/jakartaee_web_services_2_0.xsd and version 2.0

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/jaxws/wsi/w2j/rpc/literal/R2751/customfile2.xml:
```diff
@@ -17,11 +17,11 @@
 -->
 
 <jxb:bindings version="1.0"
```
should the version here change?

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/webservices13/servlet/WSMTOMFeaturesTestUsingAnnotations/custom-client.xml:
```diff
@@ -18,7 +18,7 @@
+         xmlns="https://jakarta.ee/xml/ns/jaxws"
          xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" 
          xmlns:xsd="http://www.w3.org/2001/XMLSchema" 
          xmlns:jaxb="http://java.sun.com/xml/ns/jaxb">
```
missed to update the jaxb namespace here and line 33

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/webservices13/servlet/WSMTOMFeaturesTestUsingAnnotations/custom-server.xml:
```diff
@@ -18,7 +18,7 @@
+        xmlns="https://jakarta.ee/xml/ns/jaxws"
         xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" 
         xmlns:xsd="http://www.w3.org/2001/XMLSchema" 
         xmlns:jaxb="http://java.sun.com/xml/ns/jaxb">
```
missed to update the jaxb namespace here and line 34, 40

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/webservices13/servlet/WSMTOMFeaturesTestUsingDDs/custom-client.xml:
```diff
@@ -18,7 +18,7 @@
+         xmlns="https://jakarta.ee/xml/ns/jaxws"
          xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" 
          xmlns:xsd="http://www.w3.org/2001/XMLSchema" 
          xmlns:jaxb="http://java.sun.com/xml/ns/jaxb">
```
missed to update the jaxb namespace here and line 33

Please look for more such.

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/webservices13/servlet/WSRespBindAndAddressingTestUsingAnnotations/custom-client.xml:
```diff
@@ -17,7 +17,7 @@
 -->
 
-<bindings wsdlLocation="EchoService.wsdl" xmlns="http://java.sun.com/xml/ns/jaxws">
+<bindings wsdlLocation="EchoService.wsdl" xmlns="https://jakarta.ee/xml/ns/jaxws">
```
line no. 26 can be updated to jakartaee ns

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/webservices13/servlet/WSRespBindAndAddressingTestUsingAnnotations/custom-server.xml:
```diff
@@ -17,7 +17,7 @@
 -->
 
-<bindings wsdlLocation="EchoService.wsdl" xmlns="http://java.sun.com/xml/ns/jaxws">
+<bindings wsdlLocation="EchoService.wsdl" xmlns="https://jakarta.ee/xml/ns/jaxws">
```
line 26 to be updated

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/webservices13/servlet/WSRespBindAndAddressingTestUsingDDs/custom-client.xml:
```diff
@@ -17,7 +17,7 @@
 -->
 
-<bindings wsdlLocation="EchoService.wsdl" xmlns="http://java.sun.com/xml/ns/jaxws">
+<bindings wsdlLocation="EchoService.wsdl" xmlns="https://jakarta.ee/xml/ns/jaxws">
```
please update line 26.

senivam replied:
fixed

alwin-joseph commented on src/com/sun/ts/tests/webservices13/servlet/WSRespBindAndAddressingTestUsingDDs/custom-server.xml:
```diff
@@ -17,7 +17,7 @@
 -->
 
-<bindings wsdlLocation="EchoService.wsdl" xmlns="http://java.sun.com/xml/ns/jaxws">
+<bindings wsdlLocation="EchoService.wsdl" xmlns="https://jakarta.ee/xml/ns/jaxws">
```
please update line 26.

senivam replied:
fixed

senivam commented:
Hi @alwin-joseph, thank you for the review. I've addressed your comments (and I've found more occurrences to be fixed/replaced), so please consider reviewing changes again.  Thank you :)

alwin-joseph reviewed (approved):
@senivam Thanks for the fixes. LGTM. We will trigger tck run for jaxws and platform runs for webservices12/13 after merging this to verify the results.

alwin-joseph commented:
@senivam Looks like there are build failures in platform TCK after this change 

tck build job : https://ci.eclipse.org/jakartaee-tck/job/eftl-jakartaeetck-build-900/29/
pipeline job : https://ci.eclipse.org/jakartaee-tck/blue/organizations/jenkins/jakartaee-tck/detail/master/1045/pipeline/21
console log : https://ci.eclipse.org/jakartaee-tck/blue/rest/organizations/jenkins/pipelines/jakartaee-tck/branches/master/runs/1045/nodes/21/steps/24/log/?start=0

From the logs : 
1. 
 [wsimport] [WARNING] Ignoring: binding file "file:/home/jenkins/agent/workspace/jakartaee-tck_master/src/com/sun/ts/tests/webservices13/servlet/WSWebServiceRefLookupDDs/client/customfile.xml". It is not a jaxws or a jaxb binding file.
 [wsimport]   line 20 of file:/home/jenkins/agent/workspace/jakartaee-tck_master/src/com/sun/ts/tests/webservices13/servlet/WSWebServiceRefLookupDDs/client/customfile.xml
 [wsimport] 
 [wsimport] [WARNING] Ignoring: binding file "file:/home/jenkins/agent/workspace/jakartaee-tck_master/src/com/sun/ts/tests/webservices13/servlet/WSWebServiceRefLookupDDs/client/customfile2.xml". It is not a jaxws or a jaxb binding file.
 [wsimport]   line 21 of file:/home/jenkins/agent/workspace/jakartaee-tck_master/src/com/sun/ts/tests/webservices13/servlet/WSWebServiceRefLookupDDs/client/customfile2.xml

2.  [wsimport] [WARNING] Ignoring: binding file "file:/home/jenkins/agent/workspace/jakartaee-tck_master/src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbWebServiceRefWithNoDDsTest/customfile.xml". It is not a jaxws or a jaxb binding file.
 [wsimport]   line 20 of file:/home/jenkins/agent/workspace/jakartaee-tck_master/src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbWebServiceRefWithNoDDsTest/customfile.xml

3.  [wsimport] [WARNING] Ignoring: binding file "file:/home/jenkins/agent/workspace/jakartaee-tck-alw_latest/src/com/sun/ts/tests/servlet/ee/common/sharedwebservices/servlethttptestservice/customfile.xml". It is not a jaxws or a jaxb binding file.
 [wsimport]   line 20 of file:/home/jenkins/agent/workspace/jakartaee-tck-alw_latest/src/com/sun/ts/tests/servlet/ee/common/sharedwebservices/servlethttptestservice/customfile.xml

senivam commented:
Hi @alwin-joseph, I do not think full revert is a good idea, because jakartified JAXWS (current master) will fail half of tests (it already adopts given namespaces). I can temporary revert only those files which causes failures. So it won't affect the rest of modified namespaces and we can test it VS jakartified JAXWS master

alwin-joseph commented:
> Hi @alwin-joseph, I do not think full revert is a good idea, because jakartified JAXWS (current master) will fail half of tests (it already adopts given namespaces). I can temporary revert only those files which causes failures. So it won't affect the rest of modified namespaces and we can test it VS jakartified JAXWS master

Sounds fine, you can probably revert the changes in the below directories that caused build failures. Can you also verify if the platform tck builds successful after the revert.

[dosubdirs] BUILD FAILED
[dosubdirs] The following directories failed:
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/servlet/ee/common/sharedwebservices/servlethttptestservice
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/deploy/jarDeploy
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/deploy/portcomplink/ejb
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/deploy/warDeploy
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/deploy/warMirrorSEI
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/HandlerTest/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/HandlerTest/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbAsyncTest/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbAsyncTest/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbMultipleClientInjectionTest1
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbMultipleClientInjectionTest2
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbNoWebServiceRefInClientTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbPkgEPAndClientInSameEarTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbPortFieldInjectionTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbPortMethodInjectionTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbSOAPHandlersTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbSOAPHandlersTest2
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbWebServiceProviderTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbWebServiceRefTest1
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbWebServiceRefTest2
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/annotations/WSEjbWebServiceRefWithNoDDsTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/descriptors/WSEjbOverrideWSRefHCWithDDsTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/ejb/descriptors/WSEjbOverrideWSRefWithDDsTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/narrow
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/sec/descriptors/ejb/basicSSL
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/sec/descriptors/ejb/certificate
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/sec/descriptors/servlet/basicSSL
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/sec/descriptors/servlet/certificate
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/servlet/HandlerTest/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/servlet/HandlerTest/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/servlet/WSMTOMSBAnnotationsTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/servlet/WSMTOMSBFullDDsTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/servlet/WebServiceRefsTest/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/specialcases/services/w2j/doclit/provider
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/nested1
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/nested2
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/nested3
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/nested4
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/shared1/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/shared1/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/shared2/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/shared2/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/shared3/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/shared3/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/shared4/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/shared4/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/simple1
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/simple2
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/simple3
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/simple4
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/simple5
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/simple6
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/twin1/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/twin1/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/twin2/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/twin2/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/twin3/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/file/twin4/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/nested1
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/nested2
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/nested3
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/nested4
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/shared1/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/shared1/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/shared2/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/shared2/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/shared3/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/shared3/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/shared4/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/shared4/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/simple1
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/simple2
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/simple3
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/simple4
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/simple5
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/simple6
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/twin1/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/twin1/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/twin2/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/twin2/server
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/twin3/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices12/wsdlImport/http/twin4/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/ejb/annotations/WSEjbPkgInWarUnderWebInfLibTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/ejb/annotations/WSEjbPkgInWarUnderWebInfTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/ejb/annotations/WSEjbSingletonTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/ejb/annotations/WSEjbWSRefAndAddressingCombinedTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/ejb/annotations/WSEjbWSRefRespBindAndAddressingCombinedTest
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/servlet/WSAddressingFeaturesTestUsingAnnotations
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/servlet/WSAddressingFeaturesTestUsingDDs
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/servlet/WSMTOMFeaturesTestUsingAnnotations
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/servlet/WSMTOMFeaturesTestUsingDDs
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/servlet/WSRespBindAndAddressingTestUsingAnnotations
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/servlet/WSRespBindAndAddressingTestUsingDDs
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/servlet/WSWebServiceRefLookup/client
[dosubdirs] /home/jenkins/agent/workspace/jakartaee-tck_master/bin/xml/../../src/com/sun/ts/tests/webservices13/servlet/WSWebServiceRefLookupDDs/client
[dosubdirs] 

Any idea why the errors like "Ignoring: binding file "file:/home/jenkins/agent/workspace/jakartaee-tck_master/src/com/sun/ts/tests/webservices13/servlet/WSWebServiceRefLookupDDs/client/customfile.xml". It is not a jaxws or a jaxb binding file" ?
We would want to fix that too.

alwin-joseph commented:
I also see the jaxws standalone tck has similar failures 

tck build/run job: https://ci.eclipse.org/jakartaee-tck/job/eftl-standalonetck-build-run-900/70/
pipeline job: https://ci.eclipse.org/jakartaee-tck/blue/organizations/jenkins/jakartaee-tck/detail/master/1048/pipeline/29
console log: https://ci.eclipse.org/jakartaee-tck/blue/rest/organizations/jenkins/pipelines/jakartaee-tck/branches/master/runs/1048/nodes/29/steps/32/log/?start=0

senivam commented:
actually I expect TCK for jaxws to be built only. It will fail TCK run if would be run on some staged jaxws-ri. So, when the TCK bundle with adopted namespaces will be published it shall be run only on JAXWS-ri sources in master (github) or the latest SNAPSHOT (which already contains adopted namespaces). Any staged or released JAXWS-ri will fail current TCKs because they have different namespaces.

lukasj commented:
Failures are likely caused by missing version attributtes which are mandatory now. For jaxws as well as jaxb ns, it is '3.0'. Alternatively impl can be altered to default to this version but that may require minor change to specs too.

senivam commented:
I was investigating the issue and it seems that failures occur because of GF6 which is being used in builds. 

Changes contain modified namespaces thus shall be processed by libraries which adopt those namespaces. 

This however does not happen because GF6 does not integrate them yet. 

Two libraries shall be integrated:
  - jaxb-osgi.jar and webservices-osgi.jar with jaxb-ri of minimum 3.0.0-M5 version. 
  - and JAXWS-ri of minimum 3.0.0 (20200910.184222-21) SNAPSHOT version. 

The solution could be:
- revert the whole PR and wait until pure GF6 integration
- modify GF libraries on each build (like in TCK standalone runs for each API/RI) and put required libraries on the fly.

senivam commented:
@alwin-joseph, @lukasj - any thoughts about the issue/solution?

alwin-joseph commented:
> 
>     * modify GF libraries on each build (like in TCK standalone runs for each API/RI) and put required libraries on the fly.
Can this be done simply by replacing the respective libraries in glassfish right after the gf extract for build & run of platform & jaxws tcks. We could revert this change after the GF6 integration.

Thanks for looking into it.

lukasj commented:
the idea was to make sure jaxws-ri snapshot can pass TCKs before producing 3.0.0-M5 which can be integrated into GF.

I'm not sure if simply replacing content in the jars is good way to go as this would require to also change the content of activation, mail and few apis (in the end ~5-6 jars needs to be modified to make sure tests are running within correct context)

senivam commented:
@lukasj , @alwin-joseph when I was testing TCK build locally to achieve clean build (without errors) I was replacing indicated libraries with proper content of new artifacts and build was passing. Otherwise it is not able to process new namespaces.

lukasj commented:
@senivam only make sure that all libs are in versions as they're defined at https://github.com/eclipse-ee4j/metro-jax-ws/blob/master/jaxws-ri/boms/bom/pom.xml

senivam commented:
I've created a build fix here: #510, it shall be included into main build script (I do not know where exactly to include it). Script changes only 2 affected RIs for now (jaxb and jaxws) but if it won't be enough I will add missing libraries. For now it seems to be enough to update just those two.

scottmarlow commented:
> The solution could be:
> 
>     * revert the whole PR and wait until pure GF6 integration

Another alternative to reverting the PR could be to use the [Java EE 8 schemas ](http://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/index.html#8) which technically we could use as per the Platform requirements that we use either Jakarta EE 9 schemas or schemas from the previous release (that being Jakarta EE 8 which uses the Java EE 8 schemas).

scottmarlow commented:
> I've created a build fix here: #510, it shall be included into main build script (I do not know where exactly to include it). Script changes only 2 affected RIs for now (jaxb and jaxws) but if it won't be enough I will add missing libraries. For now it seems to be enough to update just those two.

Sorry that I'm late to comment on this now.  As per previous comment, how hard would it be to switch to use Java EE 8 schemas for jaxws?

CC @smillidge

lukasj commented:
@scottmarlow does "Jakarta EE 9 WILL NOT impose any backward compatibility requirements for compatible implementations to support the Jakarta EE 8 release." from the plan mean that EE 8 descriptors must be supported by CI?

scottmarlow commented:
> @scottmarlow does "Jakarta EE 9 WILL NOT impose any backward compatibility requirements for compatible implementations to support the Jakarta EE 8 release." from the plan mean that EE 8 descriptors must be supported by CI?

@lukasj good question with regard to https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan and the text you copied.  I believe that the intent of what you copied is with regard to application backward compatibility, with the exception of application schemas.

[Jakarta EE 9 Platform Spec - Appendix A: Previous Version Deployment Descriptors](https://jakarta.ee/specifications/platform/9/platform-spec-9-SNAPSHOT.html#a3447) has the requirement:

>  This appendix describes Document Type Definitions (DTDs) and XML schemas for Deployment Descriptors from previous versions of the Java™ EE and Jakarta™ EE specifications. All Jakarta EE 9 products are required to support the DTDs and schemas specified by Jakarta EE 8, as well as the schemas specified in this version of the specification. This ensures that applications written to the previous version of the Jakarta EE specification can be deployed on products supporting the current version of this specification. Support for versions of DTDs and schemas prior to Jakarta EE 8 is optional. In addition, there are no restrictions on mixing versions of supported deployment descriptors in a single application; any combination of valid deployment descriptor versions must be supported.

lukasj commented:
@scottmarlow does available in Java SE 8 mean defined by Jakarta EE 8?

scottmarlow commented:
> @scottmarlow does available in Java SE 8 mean defined by Jakarta EE 8?
I assume you mean Java EE 8.  Yes, the schemas available via https://jakarta.ee/xml/ns/jakartaee/#8 are the schemas used by Jakarta EE 8.

Note that https://jakarta.ee/xml/ns/jakartaee/#8 links to http://xmlns.jcp.org/xml/ns/javaee#8 which currently redirects to http://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/index.html#8.

My preference is to use the brand new Jakarta EE 9 schemas, just pointing out that we could use Jakarta EE 8 schemas if needed.

senivam commented:
@scottmarlow I was pretty sure that jaxb-impl and jaxws-impl where adopting schemas from Jakarta EE 9. So, my PR was in that stream to help both implementations to pass TCK with proper Jakarta EE 9 schemas and to prove proper GF6 integration. 

However GF6 does not integrate updated implementations yet, that is why I've provided temporary fix (#510). 

Switching to Jakarta EE 8 schemas is basically what I've offered by reverting of the whole PR. The idea is that GF6 (with appropriate JAXB/JAXWS impls) shall have similar schemas as TCKs, so either do not touch it at all (and wait with merging of the gurrent PR, so revert it) or update everything (and for a while use quick and dirty solution from #510).

scottmarlow commented:
> @scottmarlow I was pretty sure that jaxb-impl and jaxws-impl where adopting schemas from Jakarta EE 9. So, my PR was in that stream to help both implementations to pass TCK with proper Jakarta EE 9 schemas and to prove proper GF6 integration.
> 

Thank you for the changes! 

> However GF6 does not integrate updated implementations yet, that is why I've provided temporary fix (#510).
> 

Thank you for providing the workaround, that is one option as you say.  

> Switching to Jakarta EE 8 schemas is basically what I've offered by reverting of the whole PR. The idea is that GF6 (with appropriate JAXB/JAXWS impls) shall have similar schemas as TCKs, so either do not touch it at all (and wait with merging of the gurrent PR, so revert it) or update everything (and for a while use quick and dirty solution from #510).

Hmm, good point, so we could revert the commit and later reapply it.

So, in summary if we apply the #510 solution we should be able to move forward and just have to revert #510 after GF later integrates with jaxws-ri update.  The downside is that we will not be testing GlassFish 6.0 in our (https://ci.eclipse.org/jakartaee-tck) TCK testing.

lukasj commented:
To make it short: what I need to be able to start ballot for XML WS is standalone TCK bundle for this. I do not care about overall platform nor GF at this point since GF is not what I'm going to use as CI, nor an environment on which the CI is going to be tested. If tests can be run and pass on GF great but that is not the goal right now for me.

scottmarlow commented:
@lukasj so this pr#503 is already merged.  Do you also need https://github.com/eclipse-ee4j/jakartaee-tck/pull/510 to be merged now?  

I think that we need to merge https://github.com/eclipse-ee4j/jakartaee-tck/pull/510 for the jaxws tck tests to (hopefully) pass via https://ci.eclipse.org/jakartaee-tck/job/eftl-standalonetck-build-run-900.

Sound right?
